lib/logthrdest/logthrdestdrv.c

```diff
@@ -1080,9 +1065,22 @@ log_threaded_dest_driver_start_workers(LogPipe *s)
 {
   LogThreadedDestDriver *self = (LogThreadedDestDriver *) s;
 
-  for (gint i = 0; i < self->num_workers; i++)
+  for (gint worker_index = 0; worker_index < self->num_workers; worker_index++)
     {
-      if (!_start_worker_thread(self, self->workers[i],i))
+      LogThreadedDestWorker *dw = self->workers[worker_index];
+
+      msg_debug("Starting dedicated worker thread",
+                evt_tag_int("worker_index", worker_index),
+                evt_tag_str("driver", self->super.super.id),
+                log_expr_node_location_tag(self->super.super.super.expr_node));
+
+      main_loop_create_worker_thread(_worker_thread,
+                                     _request_worker_exit,
+                                     dw, &self->worker_options);
+
+      _wait_for_startup_finished(dw);
+
+      if (dw->startup_failure)
```

Comment:
Just an opinion, but the original extracted method was cleaner to me.

`_start_worker_thread` could have been refactored even further, because `log_threaded_dest_driver_start_workers()` is a destination driver method, which iterates on its workers and calls their start method.
The details of how a worker is started belong to the worker, so `_start_worker_thread()` could have been a worker method (`LogThreadedDestWorker` as `self`).